src/txmempool.cpp

```diff
@@ -273,12 +561,35 @@ void CTxMemPool::removeForBlock(const std::vector<CTransaction>& vtx, unsigned i
     minerPolicyEstimator->processBlock(nBlockHeight, entries, fCurrentEstimate);
 }
 
+#if 1
+// a certificate has no conflicting txes since it has no inputs
+void CTxMemPool::removeForBlock(const std::vector<CScCertificate>& vcert, unsigned int nBlockHeight, bool fCurrentEstimate)
+{
+    LOCK(cs);
+    std::vector<CCertificateMemPoolEntry> entries;
+    for (const auto& obj : vcert)
+    {
+        uint256 hash = obj.GetHash();
+        if (mapCertificate.count(hash))
+            entries.push_back(mapCertificate[hash]);
+    }
+    for (const auto& obj : vcert)
+    {
+        std::list<std::shared_ptr<CTransactionBase>> removed;
+        remove(obj, removed, true);
+        ClearPrioritisation(obj.GetHash());
+    }
```

Comment:
As discusses AFK, there is an unhandled conflict here.
Say a valid certificate for a given epoch is in mempool, possibly with some txs spending it, and a different certificate for the very same epoch arrives in block. The block gets connected then the `removeForBlock` is called. I believe that `removeForBlock` should handle the aforementioned conflicts, as it is done for txs